### PR TITLE
Datanode remote reindex migration UI features

### PR DESCRIPTION
**Pull request by gally47**
**Datanode remote reindex migration UI features:**

- Added Log View
- Cleaned up form (shorter labels, help texts, placeholders, username & password same line)
- Made wizard navigation not clickable
- Displayed selected indices count
- Added select all / unselect all in indices list
- Added search field in indices list

Screenshot 2024-04-19 at 17 28 30

Screenshot 2024-04-29 at 17 08 59

## Motivation and Context

fixes https://github.com/Graylog2/graylog2-server/issues/18876

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by todvora — COMMENTED** [review]
Everything works fine, there is just one strange thing. Notice the selected indices counter, saying 3/11

image

Now I use the search feature and the counter displays selected counter only for the filtered results. I'd expect it to still show the overall number of selected/all, not just data for the filtered view.

image

**Review by todvora — APPROVED** [review]
LGTM! Thank you!

**Review by moesterheld — APPROVED** [review]
lgtm.
just one comment with a possible cosmetic change. could we change the color for log messages with log_level error to a light red and warnings to e.g. yellow?

File: graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/RemoteReindexRunning.tsx [hunk 1]
```diff
@@ -58,7 +73,34 @@ const RemoteReindexRunning = ({ currentStep, onTriggerStep }: MigrationStepCompo
           </IndicesContainer>
         </Alert>
       )}
-      <MigrationStepTriggerButtonToolbar nextSteps={nextSteps || currentStep.next_steps} onTriggerStep={handleTriggerStep} />
+      <MigrationStepTriggerButtonToolbar nextSteps={nextSteps || currentStep.next_steps} onTriggerStep={handleTriggerStep}>
+        <Button bsStyle="default" bsSize="small" onClick={() => setShowLogView(true)}>Log View</Button>
+      </MigrationStepTriggerButtonToolbar>
+      <BootstrapModalWrapper showModal={showLogView}
```
Comment (ousmaneo) [inline graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/RemoteReindexRunning.tsx:79]:
```suggestion
   {showLogView &&  <BootstrapModalWrapper showModal={showLogView}
```

File: graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/RemoteReindexRunning.tsx [hunk 2]
```diff
@@ -58,7 +73,34 @@ const RemoteReindexRunning = ({ currentStep, onTriggerStep }: MigrationStepCompo
...
+                    {migrationStatus.logs.map((log) => (
+                      <tr title={new Date(log.timestamp).toLocaleString()}>
+                        <td>[{log.log_level}]</td>
+                        <td>{log.message}</td>
+                      </tr>
+                    ))}
+                  </tbody>
+                </table>
+              </LogsContainer>
+            ) : ('<no logs>')}
```
Comment (ousmaneo) [inline graylog2-web-interface/src/components/datanode/migrations/remoteReindexing/RemoteReindexRunning.tsx:100]:
```suggestion
            ) : ('No logs.')}
```

**Review by ousmaneo — CHANGES_REQUESTED** [review]
It looks good; I added some small comments inline.

**Review by ousmaneo — APPROVED** [review]
LGTM!

**Pull request merged**